client/components/Assignments/AssignmentItem/fields/Headline.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+interface IProps {
+    assignment: any;
+    archiveItemForAssignment: {[assignmentId: string]: any}
+}
+
+export const HeadlineComponent = ({assignment, archiveItemForAssignment}: IProps) => {
+    const item = archiveItemForAssignment[assignment._id];
```

Comment:
It could be possible that `archiveItemForAssignment` is undefined, in which case this statement will cause an exception